contracts/v2/accounts/test/ImplementationBetaTest.sol

```diff
@@ -0,0 +1,126 @@
+ */
+
+interface AccountInterface {
+    function enable(address) external;
+    function disable(address) external;
+    function isAuth(address) external view returns (bool);
+    function isBeta() external view returns (bool);
+    function toggleBeta() external;
+}
+
+interface ConnectorsInterface {
+    function isConnectors(string[] calldata connectorNames) external view returns (bool, address[] memory);
+}
+
+contract Constants is Variables {
+    // InstaIndex Address.
+    address internal immutable instaIndex;
+    // Connectors Address.
+    address public immutable connectorsM1;
+
+    constructor(address _instaIndex, address _connectors) {
+        connectorsM1 = _connectors;
+        instaIndex = _instaIndex;
+    }
+}
+
+contract InstaImplementationBetaTest is Constants {
+
+    constructor(address _instaIndex, address _connectors) Constants(_instaIndex, _connectors) {}
+
+    function decodeEvent(bytes memory response) internal pure returns (string memory _eventCode, bytes memory _eventParams) {
+        if (response.length > 0) {
+            (_eventCode, _eventParams) = abi.decode(response, (string, bytes));
+        }
+    }
+
+    event LogCast(
+        address indexed origin,
+        address indexed sender,
+        uint256 value,
+        string[] targetsNames,
+        address[] targets,
+        string[] eventNames,
+        bytes[] eventParams
+    );
+
+    receive() external payable {}
+
+     /**
+     * @dev Delegate the calls to Connector.
+     * @param _target Connector address
+     * @param _data CallData of function.
+    */
+    function spell(address _target, bytes memory _data) internal returns (bytes memory response) {
+        require(_target != address(0), "target-invalid");
+        assembly {
+            let succeeded := delegatecall(gas(), _target, add(_data, 0x20), mload(_data), 0, 0)
+            let size := returndatasize()
+            
+            response := mload(0x40)
+            mstore(0x40, add(response, and(add(add(size, 0x20), 0x1f), not(0x1f))))
+            mstore(response, size)
+            returndatacopy(add(response, 0x20), 0, size)
+
+            switch iszero(succeeded)
+                case 1 {
+                    // throw if delegatecall failed
+                    returndatacopy(0x00, 0x00, size)
+                    revert(0x00, size)
+                }
+        }
+    }
+
+    /**
+     * @dev This is the main function, Where all the different functions are called
+     * from Smart Account.
+     * @param _targetNames Array of Connector address.
+     * @param _datas Array of Calldata.
+    */
+    function castBeta(
+        string[] calldata _targetNames,
+        bytes[] calldata _datas,
+        address _origin
+    )
+    external
+    payable 
+    returns (bytes32) // Dummy return to fix instaIndex buildWithCast function
+    {   
+        AccountInterface _dsa = AccountInterface(address(this));
+        require(_dsa.isBeta(), "Beta-does-not-enabled");
```

Comment:
```suggestion
        require(_beta, "Beta-does-not-enabled");
```

@cryptoDev222 can you check if this working? Like we are able to access the variable properly?

Reply:
Okay, got it.

Reply:
Wow, it's working!! :)